test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/Rendering/HtmlHelperDisplayExtensionsTest.cs

```diff
@@ -356,5 +427,24 @@ private class SomeModel
         {
             public string SomeProperty { get; set; }
         }
+
+        private class StatusModel
+        {
+            public Status Status { get; set; }
+        }
+
+        public class StatusResource
+        {
+            public static string Type_Faulted { get { return "Faulted from ResourceType"; } }
+        }
+
+        private enum Status : byte
+        {
+            [Display(Name = "Resx_Created")]
```

Comment:
Find both display names confusing here. Perhaps it's the fact `.resx` files are used in both `IStringLocalizer` and `ResourceType` approaches and `Type` is much broader than `ResourceType`. These lookup keys should differ from the property names but don't have to be more unique than say `"CreatedKey"` and `"FaultedKey"`.